File: src/courses/data/catalog/02-dsa/dsa-core/implementations/java/strings.java
/**
 * Java Implementation: Strings
 * Invariant: Access O(1), Search O(N), Slice O(K)
 */
package seed.dsa.core;

import java.util.*;

public class Strings<T> {
    private final List<T> items = new ArrayList<>();

    // TODO: Implement your solution here
    public void add(T value) {
        items.add(value);
    }

    public int size() {
        return items.size();
    }
}

File: src/courses/data/catalog/02-dsa/dsa-core/implementations/c/strings.c
/*
 * C Implementation: Strings
 * Invariant: Access O(1), Search O(N), Slice O(K)
 */
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int* data;
    int size;
    int capacity;
} strings_t;

// TODO: Complete the allocation and operations
void init_strings(strings_t* item, int capacity) {
    item->data = (int*)malloc(capacity * sizeof(int));
    item->size = 0;
    item->capacity = capacity;
}
